src/modules/SpaceCalculation.rb

```diff
@@ -1093,7 +1038,42 @@ def btrfs_used_size(directory)
     # where unit can be one of: "" (none) or "B", "KiB", "MiB", "GiB", "TiB", "PiB"
     # @return [Integer] size in bytes
     def size_from_string(size_str)
-      WFM.call("wrapper_storage", ["ClassicStringToByte", [size_str]])
+      classic_locale = true
+      # Assume bytes by default
+      size_str += "B" unless size_str =~ /[[:alpha:]]/
+      ::Storage.humanstring_to_byte(size_str, classic_locale)
+    end
+
+  private
+
+    # Filesystems to consider while checking the system available space
+    #
+    # @return [Array<Storage::Filesystem>]
+    def target_filesystems
+      filesystems = ::Storage::Filesystem.all(staging_devicegraph).to_a
+      filesystems.select! { |fs| fs.mountpoints.any? { |mp| mp.start_with?("/") } }
+      filesystems.reject! { |fs| TARGET_FS_TYPES_TO_IGNORE.include?(fs.type) }
+      filesystems
+    end
+
+    def staging_devicegraph
+      @storage_manager ||= Y2Packager::StorageManagerProxy.new
+      @storage_manager.staging
+    end
+
+    def filesystem_size(filesystem)
+      blk_device = filesystem.blk_devices[0]
+      # Only for local fs, NFS not supported yet in libstorage-ng
+      return 0 unless blk_device
+      blk_device.size
+    end
+
+    def filesystem_dev_name(filesystem)
+      filesystem.blk_devices[0].name
+    end
+
+    def filesystem_mountpoint(filesystem)
+      filesystem.mountpoints[0] || ""
```

Comment:
Those methods doing stuff less obvious (transversing intermediate objects or simplifying for the common case of just one device/mountpoint) are not wanted in the libstorage API (that's an already lost discussion :-) ). Using the yast2-storage-ng refinements is not an option here because we want to limit the dependency. It's not going to be like this in all the yast code.